plugins/microservices/msi_get_open_data_obj_l1desc_index/libmsi_get_open_data_obj_l1desc_index.cpp

```diff
@@ -0,0 +1,82 @@
+/// \file
+
+#include "irods_ms_plugin.hpp"
+#include "irods_re_structs.hpp"
+#include "msParam.h"
+#include "rodsErrorTable.h"
+#include "irods_error.hpp"
+#include "irods_log.hpp"
+#include "objDesc.hpp"
+#include "fileOpr.hpp"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+// persistent L1 object descriptor table
+extern l1desc_t L1desc[NUM_L1_DESC];
+
+namespace
+{
+    int msi_impl(msParam_t* _in, msParam_t* _out, ruleExecInfo_t* _rei)
+    {
+        // check sanity
+        if (!_in || !_out) {
+            return SYS_INTERNAL_ERR;
+        }
+
+        // input param needs to be a string
+        if (strcmp(_in->type, STR_MS_T)) {
+            return SYS_INVALID_INPUT_PARAM;
+        }
+
+        const char *objPath = (const char*)_in->inOutStruct;
+        int l1descInx = -1;
+
+        for (const l1desc_t &l1 : L1desc)
+        {
+            // for a valid descriptor, if the path matches...
+            if (l1.inuseFlag == FD_INUSE && !strcmp(l1.dataObjInp->objPath, objPath))
+		l1descInx = &l1 - L1desc;
+        }
+
+        // error, if nothing found
+        if (l1descInx == -1) {
+            return SYS_API_INPUT_ERR;
```

Comment:
Do we have an error code that emphasizes not found or no mapping?

Reply:
That would be great, couldn't find an obvious one myself.

Reply:
Resolved with NO_VALUES_FOUND.